### PR TITLE
Migrate the tag component to react

### PR DESCRIPTION
**Pull request by Muzych**
### 🤔 Nature of this PR

- [ ] 🐛 bug fix
- [x] ✨ new feature
- [ ] 📝 documentation improvement
- [ ] 🛝 demo code improvement
- [ ] 🚀 component style/interaction improvement
- [ ] 🏗️ ci/cd improvement
- [ ] ♻️ refactoring
- [ ] 🎨 code style optimization
- [ ] ✅ test cases
- [ ] 🔀 branch merging
- [ ] 💡 other

### 🔗 Related Issue
#7 [⭐ feature] Migrate 【Tag】 Component to React

### ✅ Pre-merge Checklist

❗️Please self-check and **check all options**.❗️

- [x] Documentation is supplemented or not needed
- [x] Code demonstration is provided or not needed
- [x] TypeScript definitions are supplemented or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: lib/components/base/tag/tag.css [hunk 1]
```diff
@@ -0,0 +1,69 @@
+:root {
+    --m-tag-h: 45px;
+    --m-tag-w-min: 80px;
+    --m-tag-border-w: calc(var(--m-tag-h) / 4.5);
+    --m-color-text-dark: #0F1423;
```
Comment (higuaifan) [inline lib/components/base/tag/tag.css]: These colors should still be placed in `global.css`. Please submit a separate PR to move `global.css` and its dependencies into the project first, and then remove `color css vars` from here;

File: lib/components/base/tag/MTag.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ */
+import React from 'react';
+import './tag.css';
+import { TagProps } from './index';
+import { MC, Slot } from '../../../types';
+import clsx from 'clsx';
+
+
+
+export default function MTag(props: TagProps & MC & Slot) {
```
Comment (higuaifan) [inline lib/components/base/tag/MTag.tsx:15]: 没有`slot`的场景下不要`&Slot`

File: lib/components/base/tag/tag.css (hunk 1 above)
Comment (Muzych) [reply lib/components/base/tag/tag.css]: image
After I try to move these color variables to global.css, it still doesn't work in production enviroment.

image

Comment (higuaifan) [reply lib/components/base/tag/tag.css]: 🤔 let me see.... how about just remove these `color css vars`, I will fix it?

Comment (Muzych) [reply lib/components/base/tag/tag.css]: Ok

Comment (higuaifan) [reply lib/components/base/tag/tag.css]: > Ok

done.
